### PR TITLE
CFM-787: Create custom "leave" page

### PR DESCRIPTION
**Pull request by DavidHernandez**
Adds a page where the user can leave the platform.

Issue: https://issuetracker.amplexor.com/jira/browse/CFM-787

In the next picture can be seen the leave link (on the tabs, as also appears on the user profile), and the form to leave the platform.

leave-form

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module [hunk 1]
```diff
@@ -213,6 +213,16 @@ function c4m_user_profile_menu() {
     'type' => MENU_CALLBACK,
   );
 
+  $items['user/%user/leave'] = array(
+    'title' => 'Leave platform',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('c4m_user_profile_remove_account_form'),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module]: `'page arguments' => array('c4m_user_profile_remove_account_form', 1),`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc [hunk 2]
```diff
@@ -45,3 +45,68 @@ function c4m_user_profile_resend_email_change_confirmation_email_form_submit($fo
...
+
+/**
+ * Form constructor for the remove account confirmation form.
+ *
+ * @see c4m_user_profile_remove_account_form_submit()
+ *
+ * @ingroup forms
+ */
+function c4m_user_profile_remove_account_form($form, $form_state) {
+  $account = menu_get_object('user', 1);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc]: remove this

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc [hunk 3]
```diff
@@ -45,3 +45,68 @@ function c4m_user_profile_resend_email_change_confirmation_email_form_submit($fo
...
 }
+
+/**
+ * Form constructor for the remove account confirmation form.
+ *
+ * @see c4m_user_profile_remove_account_form_submit()
+ *
+ * @ingroup forms
+ */
+function c4m_user_profile_remove_account_form($form, $form_state) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc]: `function c4m_user_profile_remove_account_form($form, $form_state, $account) {`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc [hunk 4]
```diff
@@ -45,3 +45,66 @@ function c4m_user_profile_resend_email_change_confirmation_email_form_submit($fo
...
+  $values = $form_state['values'];
+  $account = $form['#account'];
+
+  if ($values['anonymize']) {
+    $values['user_cancel_method'] = 'user_cancel_reassign';
+  }
+
+  user_cancel($values, $values['uid'], $values['user_cancel_method']);
+
+  $form_state['redirect'] = '/';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc]: @DavidHernandez 
Why do we redirect to `/` ?

**Comment by DavidHernandez** [comment]
The redirect to the homepage was added due to this issue: https://issuetracker.amplexor.com/jira/browse/CFM-792

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc [hunk 5]
```diff
@@ -45,3 +45,66 @@ function c4m_user_profile_resend_email_change_confirmation_email_form_submit($fo
...
+}
+
+/**
+ * Form submission handler for the remove account confirmation form.
+ *
+ * @see c4m_user_profile_remove_account_form()
+ */
+function c4m_user_profile_remove_account_form_submit($form, &$form_state) {
+  $values = $form_state['values'];
+  $account = $form['#account'];
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc]: @DavidHernandez 
It seems that the `$account` is not being used.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc (hunk 4 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.pages.inc]: @DavidHernandez 
Can you please add a comment about this one ?
thanks.

**Pull request merged**